cpp/tests/traversal/ms_bfs_test.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include "utilities/base_fixture.hpp"
+#include "utilities/check_utilities.hpp"
+#include "utilities/conversion_utilities.hpp"
+#include "utilities/test_graphs.hpp"
+#include "utilities/thrust_wrapper.hpp"
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/graph.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/utilities/high_res_timer.hpp>
+
+#include <raft/core/handle.hpp>
+#include <raft/util/cudart_utils.hpp>
+
+#include <rmm/device_scalar.hpp>
+#include <rmm/device_vector.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+#include <numeric>
+#include <random>
+#include <set>
+#include <vector>
+
+struct MsBfs_Usecase {
+  size_t radius;
+  size_t max_seeds;
+  bool test_weighted_{false};
```

Comment:
Not a must for this PR, but if you have time, you can add edge_masking option as well (similar to https://github.com/rapidsai/cugraph/blob/branch-25.06/cpp/tests/traversal/bfs_test.cpp#L86).

Reply:
Ok I did though the legacy implementation of `weakly connected components` does not support `edge masking` hence I set it to `false` for all tests

Reply:
> Have you tested this with edge masking and found that it is not supported

Yep I get an `unimplemented` error
```
unknown file: Failure
C++ exception with description "cuGraph failure at file=/home/nfs/jnke/fix_msbfs/cugraph/cpp/src/components/weakly_connected_components_impl.cuh line=1009: unimplemented.
Obtained 13 stack frames
#1 in /home/nfs/jnke/miniconda3/envs/msbfs/lib/libcugraph.so: cugraph::logic_error::logic_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) +0x5a [0x7ff385776a3a]
#2 in /home/nfs/jnke/miniconda3/envs/msbfs/lib/libcugraph.so: void cugraph::weakly_connected_components<int, int, false>(raft::handle_t const&, cugraph::graph_view_t<int, int, false, false, void> const&, int*, bool) +0x101 [0x7ff38948a051]
```

Reply:
There is an explicit check for that in WCC:`CUGRAPH_EXPECTS(!graph_view.has_edge_mask(), "unimplemented.");`

Reply:
I removed it and the test passed.